src/main/java/se/liu/ida/hefquin/engine/federation/access/impl/reqproc/GraphQLRequestProcessorImpl.java

```diff
@@ -23,17 +39,75 @@ public GraphQLRequestProcessorImpl() {
 	}
 
 	@Override
-	public JSONResponse performRequest( final GraphQLRequest req,
-	                                    final GraphQLEndpoint fm )
-	                                    		throws FederationAccessException {
-		// TODO this function needs to be implemented
-		//
-		// To get the GraphQL query that is to be executed you can do
-		// req.getGraphQLQuery() and to get the URL of the GraphQL
-		// endpoint to which the request needs to be posted you can
-		// do fm.getInterface().getURL()
-
-		return null;
-	}
+	public JSONResponse performRequest(final GraphQLRequest req,
+			final GraphQLEndpoint fm)
+			throws FederationAccessException {
+
+		final Date startTime = new Date();
+		final GraphQLQuery query = req.getGraphQLQuery();
+		final String url = fm.getInterface().getURL();
+
+		String responseBody = "{}";
+		try {
+			// Setup the connection
+			final URL endpointURL = new URL(url);
+			HttpURLConnection con = (HttpURLConnection) endpointURL.openConnection();
+			con.setRequestMethod("POST");
+			con.setConnectTimeout(connectionTimeout);
+			con.setReadTimeout(readTimeout);
+			con.setRequestProperty("Accept", "application/json");
+			con.setRequestProperty("Content-Type", "application/json");
+			con.setDoOutput(true);
+			con.connect();
+
+			// Sending post body
+			JsonObject postBody = new JsonObject();
+			postBody.put("query", query.toString());
+			postBody.put("variables", query.getParameterValues());
+			postBody.put("raw", true);
+			final OutputStreamWriter outWriter = new OutputStreamWriter(con.getOutputStream());
+			outWriter.write(postBody.toString());
+			outWriter.close();
+
+			// Fetch the input stream stream
+			InputStream iStream;
+			final int status = con.getResponseCode();
+			if (status >= 200 && status < 300) {
+				iStream = con.getInputStream();
+			} else {
+				throw new FederationAccessException(
+						"Couldn't establish a connection to endpoint. Response code: " + status, req, fm);
+			}
 
+			// Components used to read the message
+			final BufferedReader bufferReader = new BufferedReader(new InputStreamReader(iStream));
+			StringBuilder stringBuilder = new StringBuilder();
+
+			// Read from the buffer
+			String lineStr;
+			while ((lineStr = bufferReader.readLine()) != null) {
+				stringBuilder.append(lineStr);
+			}
+
+			responseBody = stringBuilder.toString();
+
+			// Disconnect
+			bufferReader.close();
+			con.disconnect();
+		} 
+		catch (IOException e) {
+			throw new FederationAccessException(req, fm);
```

Comment:
Before throwing this exception, you need to close the `bufferReader` (if there is any) and disconnect the connection.